collector/diskstats_linux.go

```diff
@@ -227,6 +309,62 @@ func (c *diskstatsCollector) Update(ch chan<- prometheus.Metric) error {
 			}
 			ch <- c.descs[i].mustNewConstMetric(val, dev)
 		}
+
+		if fsType := info[udevIDFSType]; fsType != "" {
+			ch <- c.filesystemInfoDesc.mustNewConstMetric(1.0, dev,
+				fsType,
+				info[udevIDFSUsage],
+				info[udevIDFSUUID],
+				info[udevIDFSVersion],
+			)
+		}
+
+		if name := info[udevDMName]; name != "" {
+			ch <- c.deviceMapperInfoDesc.mustNewConstMetric(1.0, dev,
+				name,
+				info[udevDMUUID],
+				info[udevDMVGName],
+				info[udevDMLVName],
+				info[udevDMLVLayer],
+			)
+		}
+
+		if ata := info[udevIDATA]; ata != "" {
+			for attr, desc := range c.ataDescs {
+				str, ok := info[attr]
+				if !ok {
+					level.Debug(c.logger).Log("msg", "Udev attribute does not exist", "attribute", attr)
+					continue
+				}
+
+				if value, err := strconv.ParseFloat(str, 64); err == nil {
+					ch <- desc.mustNewConstMetric(value, dev)
+				} else {
+					level.Error(c.logger).Log("msg", "Failed to parse ATA value", "err", err)
+				}
+			}
+		}
 	}
 	return nil
 }
+
+func udevDeviceInformation(major, minor uint32) (udevInfo, error) {
+	filename := fmt.Sprintf("/run/udev/data/b%d:%d", major, minor)
```

Comment:
This should probably be added to `paths.go` so that we use the `rootfsPath` when running in a container.

I'd almost say this is worth another flag and function like `udevFilePath` function in that file.

Also, please use the `path/filepath` package for assembling path names rather than `fmt.Sprintf()`.

Reply:
Is what I did in 662e771 what you had in mind? Or did you want the udev data path to be relative to `rootfsPath` (as in `filepath.Join(*rootfsPath, *udevDataPath)`)?

Reply:
Yes, this looks good. Thanks.